### PR TITLE
Conda recipe Lomap

### PR DESCRIPTION
**Pull request by nividic**

### CODE REVIEW

**Comment by jchodera** [comment]
(Tagging @davidlmobley):

Travis says that we are missing `pygraphviz` in one of the accessible conda channels:
```
Error:  Package missing in current linux-64 channels: 
  - pygraphviz ==1.2
```
If we search anaconda cloud for `pygraphviz` we can see that it doesn't appear in our channel (`omnia`) or in `anaconda` (the default channel):
https://anaconda.org/search?q=pygraphviz

Note that other channels have this, but it won't automatically try to retrieve the package from those channels.

To proceed, if the [PyPI 1.3.1 version](https://pypi.python.org/pypi/pygraphviz/1.3.1) works, you should use [`conda skeleton`](http://conda.pydata.org/docs/build_tutorials/pkgs.html) to create a version of the recipe in the `omnia-md/conda-recipes/` repo and check that into your PR.  Once you've done that, update your `pygraphviz` dependency in `meta.yaml`.

**Comment by nividic** [comment]
Hi there,

Lomap works with pygraphviz 1.3.1 but there are other problems to fix with anaconda. I was able to create  a recipe for pygraphviz 1.3.1 but if I try to built it a compiling error is issued. The problem is related to graphviz 2.38.0 which seems to miss a file in the include directory: graphviz/cgraph.c 

I have pasted below my recipe. I can try to push the recipe on omnia-md/conda-recipes/ but I don't think this is going to work. 

Thanks for your help

Gaetano

-------------------meta.yaml------------------

package:
  name: pygraphviz
  version: "1.3.1"

source:
  fn: pygraphviz-1.3.1.tar.gz
  url: https://pypi.python.org/packages/source/p/pygraphviz/pygraphviz-1.3.1.tar.gz
  md5: 7f690295dfe77edaa9e552d09d98d279

requirements:
  build:
    - python
    - setuptools
    - graphviz

  run:
    - python

about:
  home: http://pygraphviz.github.io
  license: BSD License
  summary: 'Python interface to Graphviz'

**Comment by rmcgibbo** [comment]
I think the "missing file" is just an indication that the pygraphviz wrappers depend on installing graphviz c library.

**Comment by jchodera** [comment]
Looks like @rmcgibbo has a fix with a `pygraphviz` recipe in #502.  We'll get that merged first and then retry the build for your `lomap` recipe.

**Comment by nividic** [comment]
Thanks a lot. Following  @rmcgibbo I was able to successfully build pygraphviz1.3.1. The problem was not in the recipe but in the build.sh file. I'm going to push a request

**Comment by rmcgibbo** [comment]
there's an infinite loop on travis, but the cause is unclear to me.

**Comment by nividic** [comment]
@rmcgibbo and @jchodera . Hi there, thanks for your help. I would like to push a new request just for my package lomap after pygraphviz is available on the omnia channel.  Would be possible to merge pygraphviz and made it available on the omnia  channel please.

**Comment by jchodera** [comment]
@rmcgibbo Looks like you would just need to skip the `win` platform and we can merge #502.

**Comment by jchodera** [comment]
I've restarted the travis build.

**Comment by jchodera** [comment]
Are you sure you don't want to add some minimal `test:` section that at least ensures `lomap` can be imported in Python (if it's a module) or a command-line function can be run successfully?  I'm not seeing any significant output from the travis build, which worries me a bit.

Here's [an example of minimal testing](https://github.com/omnia-md/conda-recipes/blob/master/yank/meta.yaml#L49-L56).

**Comment by jchodera** [comment]
Also, you will need to resolve the conflicts in your PR (eliminate the `pygraphviz` recipe).

**Comment by davidlmobley** [comment]
We were planning to add testing shortly.

On Thu, Apr 28, 2016 at 8:47 PM, John Chodera <notifications@github.com>
wrote:

> Are you sure you don't want to add some minimal test: section that at
> least ensures lomap can be imported in Python (if it's a module) or a
> command-line function can be run successfully? I'm not seeing any
> significant output from the travis build, which worries me a bit.
>
> Here's an example of minimal testing
> <https://github.com/omnia-md/conda-recipes/blob/master/yank/meta.yaml#L49-L56>
> .
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/omnia-md/conda-recipes/pull/498#issuecomment-215620120>
>



-- 
David Mobley
dmobley@gmail.com
949-385-2436

**Comment by jchodera** [comment]
> We were planning to add testing shortly.

Add testing to the recipe, or add testing to the code?

**Comment by davidlmobley** [comment]
Testing to the code, sorry for being unclear.

On Thu, Apr 28, 2016 at 8:58 PM, John Chodera <notifications@github.com>
wrote:

> We were planning to add testing shortly.
>
> Add testing to the recipe, or add testing to the code?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/omnia-md/conda-recipes/pull/498#issuecomment-215621016>
>



-- 
David Mobley
dmobley@gmail.com
949-385-2436

**Comment by jchodera** [comment]
I think that would be a good thing to add *prior* to distributing it.  Can we help point you to some examples?

**Comment by nividic** [comment]
It is still looping in Travis and I don't know why. I added the test section as well

**Comment by nividic** [comment]
By the way in my lomap github project the travis [tests](https://travis-ci.org/nividic/Lomap) are passed without any loop

**Comment by nividic** [comment]
Hi there, 
How can I tell travis to skip the testing of python 3.x?
Thanks

**Comment by jchodera** [comment]
You can find an example [here](https://github.com/omnia-md/conda-recipes/blob/master/assaytools/meta.yaml#L14-L15).

**Comment by nividic** [comment]
Thanks. Please DO NOT merge because I'm having issue with the OS X version

**Comment by jchodera** [comment]
We typically mark the pull request title with `[WIP]` to indicate it is still "work in progress". I've done that for you now, but you can change it back when it is ready to review and merge.

**Comment by nividic** [comment]
Thanks a lot I didn't know that. I found the problem with OS X. The graphviz  version 2.38.0-1 is broken and it doesn't work on mac. In a fresh installation of miniconda I installed graphviz from the default channel. I tried to run dot (a graphviz program) and I got the error:  

dyld: Library not loaded: /usr/lib/libltdl.7.dylib
  Referenced from: /Users/gac/local/miniconda2/envs/test/bin/dot
  Reason: image not found
Trace/BPT trap: 5

The graphviz version 2.38.0-3 works on mac but is not present in the default channel. I have to create a recipe and add it to omnia.

**Comment by jchodera** [comment]
Either that or we could just copy the package to `omnia`.  Let me know if the recipe creation proves to be difficult and we will fall back on that if needed.

**Comment by nividic** [comment]
Hi there,

I was able to build graphviz 2.38.0-3 and all the travis tests are passed. I have some questions as well. Pygraphviz depends on graphviz and I wonder if we need to rebuild pygraphviz against the new version of graphviz. In addition if I'll chamge the lomap code in future do I need to resubmit to the omnia-recipe a new PR to build the new omina channel lomap version?

Thanks

**Comment by rmcgibbo** [comment]
> Pygraphviz depends on graphviz and I wonder if we need to rebuild pygraphviz against the new version of graphviz.

Probably, yes.

> In addition if I'll change the lomap code in future do I need to resubmit to the omnia-recipe a new PR to build the new lomap version?

Also, yes.

**Comment by nividic** [comment]
Thanks for all the comments and help.

I think that we can try to merge please. In addition also pygraphviz should be rebuild.

File: graphviz/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  fn: graphviz-2.38.0.tar.gz
+  url: http://www.graphviz.org/pub/graphviz/stable/SOURCES/graphviz-2.38.0.tar.gz
+  md5: 5b6a829b2ac94efcd5fa3c223ed6d3ae
+  patches:
+     - gvconfig_libdir.patch
+
+build:
+  number: 3
+  binary_relocation: true
+
```
Comment (jchodera) [inline graphviz/meta.yaml]: I think you need to skip this on windows by adding:
```
  skip:
    - [py3k]
```

**Comment by jchodera** [comment]
`appveyor` is failing.  Can you have the build skip trying to build your recipe on `win`?

I've noted where you have to add a `skip` to the recipe.

**Comment by jchodera** [comment]
We're good now, thanks!

**Comment by nividic** [comment]
Hi there,

I cannot found the lomap and graphviz packages on omnia by using conda (e.g. conda search lomap). I have correctly added the omnia channel to my .condarc file. I was checking on anaconda cloud ad both lomap and graphviz have been generated but there is a warning:

Warning: This package does not contain any files in the main label. Contact the package admin for developer install instructions. 

Can someone help with this please? What I did wrong?

Thanks

Gaetano

**Comment by mpharrigan** [comment]
Hi @nividic 
Our new QA procedure sends all new builds to a special "label" called `rc` for release candidate. Users will only get this with

```
conda install -c omnia/labels/rc lomap
```

Once you verify that it built correctly, you can notify us and we'll move it to the `main` omnia label

**Comment by nividic** [comment]
Thanks,

There are problems on Mac. @rmcgibbo would be possible to rebuild pygraphviz against the new graphviz version please?

**Comment by nividic** [comment]
The problem seems to be still in graphviz on Mac

**Comment by nividic** [comment]
@jchodera would it be possible to copy from my anaconda repository ([https://anaconda.org/nividic/graphviz/files](url) the graphviz osx package (osx-64/graphviz-2.38.0-3.tar.bz2) into the related omnia graphviz package please?

**Comment by rmcgibbo** [comment]
The best thing is just to submit a pull request to build the new recipe version through omnia/conda-recipes. Is there a reason that won't work?

**Comment by nividic** [comment]
@rmcgibbo I did. graphviz has been generated and available on omnia (rc release)  but for some reasons the mac version does not work. I would like to see if my version works with omnia (it would be  just a test).

**Comment by nividic** [comment]
For some unclear reason graphviz on omnia is built adding an extra dynamic library dependence in the graphviz "dot" file executable in OS X. Calling ottool -L dot 

produces:
	
@rpath/libgvc.6.dylib (compatibility version 7.0.0, current version 7.0.0)
/usr/local/lib/libltdl.7.dylib (compatibility version 11.0.0, current version 11.0.0)
@rpath/libxdot.4.dylib (compatibility version 5.0.0, current version 5.0.0)
@rpath/libcgraph.6.dylib (compatibility version 7.0.0, current version 7.0.0)
@rpath/libpathplan.4.dylib (compatibility version 5.0.0, current version 5.0.0)
/usr/lib/libexpat.1.dylib (compatibility version 7.0.0, current version 7.2.0)
@rpath/libz.1.dylib (compatibility version 1.0.0, current version 1.2.8)
/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)
@rpath/libcdt.5.dylib (compatibility version 6.0.0, current version 6.0.0)

on a working copy of graphviz there is no dependence at all from:
 
/usr/local/lib/libltdl.7.dylib (compatibility version 11.0.0, current version 11.0.0)


The recipe that I'm using to build graphviz on omnia and on my workstation is the same.

**Comment by rmcgibbo** [comment]
Can you add a test to the `test` section of the recipe that actually _uses_ dot, so that we can verify whether it is working correctly when its built?

**Comment by nividic** [comment]
@rmcgibbo 

It is already [there](https://github.com/omnia-md/conda-recipes/blob/master/graphviz/meta.yaml)

**Comment by rmcgibbo** [comment]
Perhaps we're miscommunicating a little bit.

My understanding from what you've written is that the conda recipe and the resulting packages in this repository are somehow broken. And yet, the recipe seems to build on travis (and pass the test) just fine. So my suspicion was that perhaps the test inside the recipe is wholly insufficient.

I have no special insight into this problem, having never used graphviz.

**Pull request merged**